### ISSUE TITLE
`sqlcmd open ads` fails with: Not enough memory resources are available to process this command"

### ISSUE THREAD

**Issue by stuartpa** [microsoft/go-sqlcmd#233]
A Bug Bash tester hit this error when opening ADS, even though there is 10GB free on the machine

It might be this issue with too many credentials in credman:

https://github.com/desktop/desktop/issues/15217

**Comment by SQLDBAWithABeard** [comment]
I fixed this by removing credentials using the CredentialManager module

```
 $a = Get-StoredCredential  -AsCredentialObject
 $a | Where UserName -Like '*Ben*' | ForEach-Object { Remove-StoredCredential -Target $_.TargetName}
```

Then I was able to open ADS
![image](https://user-images.githubusercontent.com/6729780/218791354-a5acf164-85f8-41ff-8e13-d78ecda1db4b.png)

**Comment by shueybubbles** [comment]
Is Azure Data Studio "leaking" credentials? Ie if you create a new connection in ADS and it adds the credential, when you delete the connection does it delete the credential?

**Comment by shueybubbles** [comment]
Indeed...https://github.com/microsoft/azuredatastudio/issues/21936

**Comment by cheenamalhotra** [comment]
Credential limit is applicable even when you do delete credentials but needed to save more than a certain number of passwords in Credential Store.. This is also reproducible CredentialManager PS module (see https://github.com/microsoft/azuredatastudio/issues/20990#issuecomment-1293041067 for more info).

**Comment by shueybubbles** [comment]
For v1 - detect the error and give a better message.

**Comment by stuartpa** [comment]
Fixed (better error message added) in #240
